foundationdb/datadog_checks/foundationdb/check.py

```diff
@@ -29,7 +31,13 @@ def construct_database(self):
             fdb.options.set_tls_verify_peers(self.instance.get('tls_verify_peers').encode('latin-1'))
 
         if 'cluster_file' in self.instance:
-            self._db = fdb.open(cluster_file=self.instance.get('cluster_file'))
+            cluster_file = self.instance.get('cluster_file')
+
+            if self.instance.get('copy_cluster_file'):
+                _, cluster_file = tempfile.mkstemp(suffix=".cluster")
```

Comment:
Is there a "right place" to clean up this temporary file (or other long-lived resources)? I notice that this check doesn't have a mechanism for closing the FoundationDB client, so perhaps just leaving the temporary file is the right thing to do.

Reply:
I'm sorry to turn this around into a request for basic information, but I've actually really been struggling to find a clear articulation of the lifecycle of cluster checks in particular. Can you point me to the relevant docs or, failing that, tell me a little more about the lifecycle of a cluster check/its parent Python environment?

Reply:
Ah thank you, I see now, that makes a lot of sense. I was not familiar enough with the check but if the connection is persisted then we will not need to clean up the file on every check run.